src/contractTest/java/uk.gov.hmcts.reform.idam.client/IdamClientConsumerTest.java

```diff
@@ -0,0 +1,94 @@
+package uk.gov.hmcts.reform.idam.client;
+
+import au.com.dius.pact.consumer.Pact;
+import au.com.dius.pact.consumer.dsl.PactDslJsonBody;
+import au.com.dius.pact.consumer.dsl.PactDslWithProvider;
+import au.com.dius.pact.consumer.junit5.PactConsumerTestExt;
+import au.com.dius.pact.consumer.junit5.PactTestFor;
+import au.com.dius.pact.model.RequestResponsePact;
+import com.google.common.collect.Maps;
+import org.json.JSONException;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.cloud.openfeign.EnableFeignClients;
+import org.springframework.context.annotation.PropertySource;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatus;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+
+import java.util.Map;
+import java.util.TreeMap;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@EnableFeignClients(basePackages = {"uk.gov.hmcts.reform.idam.client"})
+@PropertySource(value = "classpath:application.yml")
+@EnableAutoConfiguration
+@ExtendWith(PactConsumerTestExt.class)
+@ExtendWith(SpringExtension.class)
+@PactTestFor(providerName = "Idam_api", port = "5050")
+@SpringBootTest(classes = {IdamClient.class})
+public class IdamClientConsumerTest {
+
+    @Autowired
+    private IdamClient idamClient;
+
+    private static final String IDAM_OPENID_TOKEN_URL = "/o/token";
+
+    @Pact(provider = "Idam_api", consumer = "idamClient")
+    public RequestResponsePact executeGetIdamAccessTokenAndGet200(PactDslWithProvider builder) throws JSONException {
+
+        String[] rolesArray = new String[1];
+        rolesArray[0] = "citizen";
+        Map<String, String> requestheaders = Maps.newHashMap();
+        requestheaders.put("Content-Type", "application/x-www-form-urlencoded");
+
+        Map<String, String> responseheaders = Maps.newHashMap();
+        responseheaders.put("Content-Type", "application/json");
+
+        Map<String, Object> params = new TreeMap<>(String.CASE_INSENSITIVE_ORDER);
+        params.put("email", "emCaseOfficer@email.net");
+        params.put("password", "Password123");
+        params.put("forename","emCaseOfficer");
+        params.put("surname", "jar123");
+        params.put("roles", rolesArray);
+
+        return builder
+                .given("a user exists", params)
+                .uponReceiving("Provider takes user/pwd and returns Auth code to Idam Client")
+                .path(IDAM_OPENID_TOKEN_URL)
+                .method(HttpMethod.POST.toString())
+                .body("redirect_uri=https%3A%2F%2Flocalhost%3A5000%2Freceiver&client_id=bsp&grant_type=password"
```

Comment:
will this work when idam tries to verify the pact?

Reply:
Now working when running Pact provider verify for idam-api.

![image](https://user-images.githubusercontent.com/4736395/84872640-10a57900-b07a-11ea-892a-3122e7814cdd.png)

Reply:
Failed 23 hours ago but maybe need idam to re-run verification?

![image](https://user-images.githubusercontent.com/21194782/84909272-61829500-b0ad-11ea-8b32-ca0957782484.png)

Reply:
So the above was running provider pact verification locally against idam-api. The test I've added is on par with the em-annotations one which is passing. Based on that I'm confident the test as of this today is correct.

Making publishing a consumer pact test PR dependent dependent on provider running and verifying against prod pipeline definitely feels like a bad pattern. If someone wants to extend a provider verification step for consumer prs that's fine but beyond scope of this PR